File: client-scalaz/src/main/scala/org/openapitools/client/api/Flag.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Flag._

case class Flag (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
owner: Option[Account],
flagableObjectType: Option[String],
flagableObjectId: Option[Long],
flagDescription: Option[String],
app: Option[String])

object Flag {
  import DateTimeCodecs._

  implicit val FlagCodecJson: CodecJson[Flag] = CodecJson.derive[Flag]
  implicit val FlagDecoder: EntityDecoder[Flag] = jsonOf[Flag]
  implicit val FlagEncoder: EntityEncoder[Flag] = jsonEncoderOf[Flag]
}

File: client-clojure/src/sirqul_io_t_platform/specs/flag.clj
(ns sirqul-io-t-platform.specs.flag
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account :refer :all]
            )
  (:import (java.io File)))


(def flag-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :owner) account-spec
   (ds/opt :flagableObjectType) string?
   (ds/opt :flagableObjectId) int?
   (ds/opt :flagDescription) string?
   (ds/opt :app) string?
   })

(def flag-spec
  (ds/spec
    {:name ::flag
     :spec flag-data}))
